services/web/server/src/simcore_service_webserver/meta_handlers.py

```diff
@@ -0,0 +1,277 @@
+                    tag.name, return_none_if_fails=True
+                ):
+                    if iteration:
+                        raise ValueError(
+                            f"This entry has more than one iteration {tag=}"
+                        )
+                    iteration = pim
+                elif pid := parse_workcopy_project_tag_name(tag.name):
+                    if workcopy_id:
+                        raise ValueError(
+                            f"This entry has more than one workcopy  {tag=}"
+                        )
+                    workcopy_id = pid
+                else:
+                    log.debug("Got %s for children of %s", f"{tag=}", f"{commit_id=}")
+
+            if not workcopy_id:
+                raise ValueError("No working copy found")
+            if not iteration:
+                raise ValueError("No iteration tag found")
+
+            iterations.append((workcopy_id, iteration.iter_index))
+
+        except ValueError as err:
+            log.debug("Skipping child %s: %s", n, err)
+
+    total_number_of_iterations = len(iterations)
+
+    # sort and select. If requested interval is outside of range, it returns empty
+    iterations.sort(key=lambda tup: tup[1])
+
+    if limit is None:
+        return iterations[offset:], total_number_of_iterations
+
+    return iterations[offset : (offset + limit)], total_number_of_iterations
+
+
+async def create_or_get_project_iterations(
+    vc_repo: VersionControlForMetaModeling,
+    project_uuid: ProjectID,
+    commit_id: CommitID,
+) -> List[IterationTuple]:
+
+    raise NotImplementedError()
+
+
+# MODELS ------------------------------------------------------------
+
+
+class ParentMetaProjectRef(BaseModel):
+    project_id: ProjectID
+    ref_id: CheckpointID
+
+
+class ProjectIterationAsItem(BaseModel):
+    name: str = Field(
+        ...,
+        description="Iteration's resource name [AIP-122](https://google.aip.dev/122)",
+    )
+    parent: ParentMetaProjectRef = Field(
+        ..., description="Reference to the the meta-project that created this iteration"
+    )
+
+    workcopy_project_id: ProjectID = Field(
+        ...,
+        description="ID to this iteration's working copy."
+        "A working copy is a real project where this iteration is run",
+    )
+
+    workcopy_project_url: HttpUrl = Field(
+        ..., description="reference to working copy project"
+    )
+    url: HttpUrl = Field(..., description="self reference")
+
+
+# ROUTES ------------------------------------------------------------
+
+
+routes = web.RouteTableDef()
+
+
+@routes.get(
+    f"/{VTAG}/projects/{{project_uuid}}/checkpoint/{{ref_id}}/iterations",
+    name=f"{__name__}._list_meta_project_iterations_handler",
+)
+@permission_required("project.snapshot.read")
+async def _list_meta_project_iterations_handler(request: web.Request) -> web.Response:
+    # FIXME: check access to non owned projects user_id = request[RQT_USERID_KEY]
```

Comment:
again is this a current bug in the implementation? or a missing feature? or something forgotten?
If an issue maybe link the issue that goes with it.

Reply:
added reference to #2735